trieste/models/config.py

```diff
@@ -13,14 +13,28 @@
 # limitations under the License.
 from __future__ import annotations
 
+from collections.abc import Callable
 from dataclasses import dataclass, field
 from typing import Any, Dict, Union
 
 import gpflow
 import tensorflow as tf
-
-from .model_interfaces import TrainableProbabilisticModel, supported_models
-from .optimizer import create_optimizer
+from gpflow.models import GPR, SGPR, VGP
+
+from .interfaces import TrainableProbabilisticModel
+from .optimizer import Optimizer, create_optimizer
+from .models_gpflow.models import GaussianProcessRegression, VariationalGaussianProcess
+
+
+"""
+A mapping of third-party model types to :class:`CustomTrainable` classes that wrap models of those
+types.
+"""
+supported_models: dict[Any, Callable[[Any, Optimizer], TrainableProbabilisticModel]] = {
+    GPR: GaussianProcessRegression,
+    SGPR: GaussianProcessRegression,
+    VGP: VariationalGaussianProcess,
+}
 
 
 def _default_optimizer() -> gpflow.optimizers.Scipy:
```

Comment:
I think it would be good to eliminate gpflow here, but what to use instead, tensorflow? since that is a library used throughout?

Reply:
do you mean for the optimizer?

Reply:
yes, text refers to `_default_optimizer`, while I marked GPR, SGPR and VGP which is also an import from gpflow

Reply:
i'd always thought of this file as gpflow-specific

Reply:
ok, then probably I misunderstood how this config is used, thanks for the tip, that would resolve these issues effectively